src/sqlfluff/core/parser/segments/raw.py

```diff
@@ -58,12 +58,13 @@ def __init__(
         self._source_fixes = source_fixes
         # UUID for matching
         self.uuid = uuid or uuid4()
-
-    def __repr__(self):
-        return "<{}: ({}) {!r}>".format(
+        self.representation = "<{}: ({}) {!r}>".format(
             self.__class__.__name__, self.pos_marker, self.raw
         )
 
+    def __repr__(self):
+        return self.representation
+
```

Comment:
Looks great 👍 . Can you add a short comment here to note the intent of `self.representation` for future-us? Feels like something we'll totally forget.

Reply:
Done 👍